bigtable/beam/bulk-data-generator/src/main/java/bigtable/BulkWrite.java

```diff
@@ -0,0 +1,201 @@
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.joda.time.Duration;
+
+public class BulkWrite {
+
+  static final double TB_PER_TABLE = .5;
+  static final long MB_PER_ROW = 1;
+  static final long ONE_MB = 1000 * 1000;
+  static final long ONE_GB = 1000 * ONE_MB;
+  static final long ONE_TB = 1000 * ONE_GB;
+  static final long MB_PER_SEC = 120;
+
+  static final String COLUMN_FAMILY = "cf";
+  public static final String TABLE_PREFIX = "data-";
+  static final SecureRandom random = new SecureRandom();
+
+  public static void main(String[] args) {
+    BigtableOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(BigtableOptions.class);
+    Pipeline p = Pipeline.create(options);
+
+    try {
+      BigtableTableAdminSettings adminSettings = BigtableTableAdminSettings.newBuilder()
+          .setProjectId(options.getProject())
+          .setInstanceId(options.getBigtableInstanceId())
+          .build();
+
+      BigtableTableAdminClient adminClient = BigtableTableAdminClient.create(adminSettings);
+
+      int clusterNodeCount = getClusterNodeCount(options.getProject(),
+          options.getBigtableInstanceId());
+      List<String> newTableIds = getNewTableIds(adminClient, options.getBigtableSize());
+
+      if (newTableIds.isEmpty()) {
+        return;
+      }
+
+      long numRows = (long) ((TB_PER_TABLE * ONE_TB) / (MB_PER_ROW * ONE_MB));
+      long rate = clusterNodeCount * MB_PER_SEC / newTableIds.size();
+
+      String generateLabel = String
+          .format("Generate %d rows at %dMB per second for %d tables", numRows, rate,
+              newTableIds.size());
+      String mutationLabel = String
+          .format("Create mutations that write %d MB to each row", MB_PER_ROW);
+
+      System.out.println(generateLabel);
+      System.out.println(mutationLabel);
+
+      PCollection<Mutation> mutations = p
+          .apply(generateLabel, GenerateSequence.from(0).to(numRows)
+              .withRate(rate, Duration.standardSeconds(1)))
+          .apply(mutationLabel,
+              ParDo.of(new CreateMutationFn()));
+
+      for (String tableId : newTableIds) {
+        mutations.apply(String.format("Write data to table %s", tableId),
+            CloudBigtableIO.writeToTable(new CloudBigtableTableConfiguration.Builder()
+                .withProjectId(options.getProject())
+                .withInstanceId(options.getBigtableInstanceId())
+                .withTableId(tableId)
+                .build()));
+      }
+
+      p.run();
+    } catch (IOException e) {
+      System.out.println("Unable to connect to Bigtable instance.");
+    }
+  }
+
+  private static List<String> getNewTableIds(BigtableTableAdminClient adminClient,
```

Comment:
Since this code isn't provided inline with the sample, please add a `//` comment describing what it does.